src/coreclr/vm/gc_unwind_x86.inl

```diff
@@ -0,0 +1,3857 @@
+            curOffs = (unsigned)pFirstBaseSPslot[1] - 1;
+            _ASSERTE(curOffs < info.methodSize);
+
+            /* Extract the necessary information from the info block header */
+
+            table = PTR_CBYTE(gcInfoToken.Info);
+
+            table += DecodeGCHdrInfo(gcInfoToken,
+                                     curOffs,
+                                     &info);
+        }
+    }
+#endif
+
+    bool        willContinueExecution = !(flags & ExecutionAborted);
+    unsigned    pushedSize = 0;
+
+    /* if we have been interrupted we don't have to report registers/arguments
+     * because we are about to lose this context anyway.
+     * Alas, if we are in a ebp-less method we have to parse the table
+     * in order to adjust ESP.
+     *
+     * Note that we report "this" for all methods, even if
+     * noncontinuable, because of the off chance they may be
+     * synchronized and we have to release the monitor on unwind. This
+     * could conceivably be optimized, but it turns out to be more
+     * expensive to check whether we're synchronized (which involves
+     * consulting metadata) than to just report "this" all the time in
+     * our most important scenarios.
+     */
+
+    if  (info.interruptible)
+    {
+        unsigned curOffsRegs = curOffs;
+
+        // Don't decrement curOffsRegs when it is 0, as it is an unsigned and will wrap to MAX_UINT
+        //
+        if (curOffsRegs > 0)
+        {
+            // If we are not on the active stack frame, we need to report gc registers
+            // that are live before the call. The reason is that the liveness of gc registers
+            // may change across a call to a method that does not return. In this case the instruction
+            // after the call may be a jump target and a register that didn't have a live gc pointer
+            // before the call may have a live gc pointer after the jump. To make sure we report the
+            // registers that have live gc pointers before the call we subtract 1 from curOffs.
+            if ((flags & ActiveStackFrame) == 0)
+            {
+                // We are not the top most stack frame (i.e. the ActiveStackFrame)
+                curOffsRegs--;   // decrement curOffsRegs
+            }
+        }
+
+        pushedSize = scanArgRegTableI(skipToArgReg(info, table), curOffsRegs, curOffs, &info);
+
+        RegMask   regs  = info.regMaskResult;
+        RegMask  iregs  = info.iregMaskResult;
+        ptrArgTP  args  = info.argMaskResult;
+        ptrArgTP iargs  = info.iargMaskResult;
+
+        _ASSERTE((isZero(args) || pushedSize != 0) || info.ebpFrame);
+        _ASSERTE((args & iargs) == iargs);
+        // Only synchronized methods and generic code that accesses
+        // the type context via "this" need to report "this".
+        // If its reported for other methods, its probably
+        // done incorrectly. So flag such cases.
+#ifndef FEATURE_NATIVEAOT
```

Comment:
I think it would be fine to drop this assert to get rid of FEATURE_NATIVEAOT ifdefs.

Reply:
Done